src/patchkernel/surface_skd_tree.cpp

```diff
@@ -581,6 +490,226 @@ long SurfaceSkdTree::findPointClosestCell(int nPoints, const std::array<double,
+* distance
+* \param[in] interiorCellsOnly if set to true, only interior cells will be considered,
+* it will be possible to consider non-interior cells only if the tree has been
+* instantiated with non-interior cells support enabled
+* \param[out] ids on output it will contain the ids of the closest cells.
+* If all cells contained in the tree are farther than the maximum
+* distance, the first element of the container will be set to the null id.
+* \param[out] distance on output it will contain the distance between
+* the point and closest cell. If all cells contained in the tree are
+* farther than the maximum distance, the argument will be set to the
+* maximum representable distance
+*/
+long SurfaceSkdTree::findPointClosestCells(const std::array<double, 3> &point, double maxDistance,
+                                          bool interiorCellsOnly, std::vector<long> &ids, double *distance) const
+{
+    // Tolerance for distance evaluations
+    const PatchKernel &patch = getPatch();
+    double tolerance = patch.getTol();
+
+    // Initialize the cell id
+    ids.resize(1, Cell::NULL_ID);
+
+    // Get the root of the tree
+    std::size_t rootId = 0;
+    const SkdNode &root = m_nodes[rootId];
+    if (root.isEmpty()) {
+        *distance = std::numeric_limits<double>::max();
+
+        return 0;
+    }
+
+    Candidates candidates;
+    findPointClosestCandidates(point, maxDistance, distance, candidates);
+
+    // Process the candidates and find the closest cell
+    long nDistanceEvaluations = 0;
+    for (std::size_t k = 0; k < candidates.candidateIds->size(); ++k) {
+        // Do not consider nodes with a minimum distance greater than the
+        // distance estimate
+        if (utils::DoubleFloatingGreater()(candidates.candidateMinDistances->at(k), *distance, tolerance, tolerance)) {
+            continue;
+        }
+
+        // Evaluate the distance
+        std::size_t nodeId = candidates.candidateIds->at(k);
+        const SkdNode &node = m_nodes[nodeId];
+
+        node.updatePointClosestCells(point, interiorCellsOnly, ids, distance);
+        ++nDistanceEvaluations;
+    }
+
+    return nDistanceEvaluations;
+}
+
+/*!
+* Given the specified point find the closest candidate cells contained
+* in the tree and estimate the minimum distance between that cells and
+* the given point.
+*
+* \param[in] point is the point
+* \param[in] maxDistance all cells whose distance is greater than
+* this parameters will not be considered for the evaluation of the
+* distance
+* \param[out] distanceEstimate on output it will contain the estimate
+* of the distance between the point and closest cell. If all cells
+* contained in the tree are farther than the maximum distance, the
+* argument will be set to the maximum representable distance
+*/
+void SurfaceSkdTree::findPointClosestCandidates(const std::array<double, 3> &point, double maxDistance,
+        double *distanceEstimate, Candidates &candidates) const
```

Comment:
Since the Candidates already contain information about the distance of the candidates, we can move the distance estimate into the structure. We can call it maximumDistance.

Reply:
Or maxDistance to be coherent with the other member of the distance.